src/config/rcxml.c

```diff
@@ -119,61 +119,145 @@ parse_window_type(const char *type)
+ * Instead of openbox's <titleLayout>WLIMC</title> we use
+ *
+ *     <titlebar>
+ *       <layout>menu:iconfiy,max,close</layout>
+ *       <showTitle>yes|no</showTitle>
+ *     </titlebar>
+ *
+ * ...using the icon names (like iconify.xbm) without the file extension for the
+ * identifier.
+ *
+ * labwc        openbox     description
+ * -----        -------     -----------
+ * menu         W           Open window menu (client-menu)
+ * iconfiy      I           Iconify (aka minimize)
+ * max          M           Maximize toggle
+ * close        C           Close
+ * shade        S           Shade toggle
+ * desk         D           All-desktops toggle (aka omnipresent)
+ */
 static void
-fill_title_layout(char *nodename, char *content)
+fill_section(const char *content, struct wl_list *list)
 {
-	char *c, *c2;
-	enum ssd_part_type type;
-	struct title_button *item;
-	struct wl_list *list = &rc.title_buttons_left;
-
-	for (c = content; *c != '\0'; c++) {
-		for (c2 = content; c2 < c; c2++) {
-			if (*c2 == *c) {
-				break;
-			}
-		}
-		if (c2 != c) {
+	gchar **identifiers = g_strsplit(content, ",", -1);
+	for (size_t i = 0; identifiers[i]; ++i) {
+		char *identifier = identifiers[i];
+		if (string_null_or_empty(identifier)) {
 			continue;
 		}
-
-		switch (*c) {
-		/* case 'N': icon */
-		case 'L':
-			list = &rc.title_buttons_right;
-			rc.show_title = true;
-			continue;
-		case '|':
-			list = &rc.title_buttons_right;
-			continue;
-		case 'W':
+		enum ssd_part_type type = LAB_SSD_NONE;
+		if (!strcmp(identifier, "menu")) {
 			type = LAB_SSD_BUTTON_WINDOW_MENU;
-			break;
-		case 'I':
+		} else if (!strcmp(identifier, "iconify")) {
 			type = LAB_SSD_BUTTON_ICONIFY;
-			break;
-		case 'M':
+		} else if (!strcmp(identifier, "max")) {
 			type = LAB_SSD_BUTTON_MAXIMIZE;
-			break;
-		case 'C':
+		} else if (!strcmp(identifier, "close")) {
 			type = LAB_SSD_BUTTON_CLOSE;
-			break;
-		case 'S':
+		} else if (!strcmp(identifier, "shade")) {
 			type = LAB_SSD_BUTTON_SHADE;
-			break;
-		case 'D':
+		} else if (!strcmp(identifier, "desk")) {
 			type = LAB_SSD_BUTTON_OMNIPRESENT;
-			break;
-		default:
+		} else {
+			wlr_log(WLR_ERROR, "invalid titleLayout identifier '%s'",
+				identifier);
 			continue;
 		}
 
-		item = znew(*item);
+		assert(type != LAB_SSD_NONE);
+
+		struct title_button *item = znew(*item);
 		item->type = type;
 		wl_list_append(list, &item->link);
 	}
+	g_strfreev(identifiers);
+}
+
+static int
+compare_strings(const void *a, const void *b)
+{
+	char * const *str1 = a;
+	char * const *str2 = b;
+	return strcmp(*str1, *str2);
+}
+
+static bool
+contains_duplicates(char *content)
+{
+	bool ret = false;
+	char *p, *s = xstrdup(content);
+
+	/*
+	 * The string typically looks like: 'menu:iconfiy,max,close' so we
+	 * replace colons with commas and then split on commas to get a simple
+	 * string list of all the identifiers.
+	 */
+	while ((p = strchr(s, ':'))) {
+		*p = ',';
+	}
+	gchar **idents = g_strsplit(s, ",", -1);
```

Comment:
I think `g_strsplit_set()` can be used here to split a string with multiple delimiters.

Reply:
Nice. Didn't know about that one. Done.